website/content/blog/Towards-Optimized-Microservices-Performance-Sustainability-via-Istio-Kepler-and-Smart-Scheduling.md

```diff
@@ -0,0 +1,64 @@
+
+Scenario 2: Customized Service Allocation
+
+In contrast, the second scenario adopts a more targeted approach. Here, the "Data Store" Service is strategically placed on Worker01, while the "Business Logic" Service finds its home on Worker02. This customization allows for meticulous resource allocation, aligning services with nodes that best suit their computational and memory needs.
+
+![Typical Scheduling Policy](images/241-scheduling.png)
+
+In terms of performance, there were notable differences between the scenarios. In the default Kubernetes scenario, the P99 latency and transactions per second (TPS) were superior compared to the second scenario. However, when considering energy consumption, the graph displayed the energy consumption trends of all pods within the hotel-res namespace over time. Notably, the energy consumption between the two scenarios remained relatively similar.
+
+Later on, Smart Scheduling with Reinforcement Learning is introduced: Firstly, it observes various metrics such as workloads, Istio and Kepler metrics, CPU usage, and memory utilization. Based on this information, it devises scheduling policies for pods across different nodes, adjusting configurations accordingly. These decisions are guided by performance and energy consumption evaluations, forming the basis of rewards received by the system.
+
+In the Kubernetes cluster environment, Smart Scheduling executes by scheduling pods onto nodes as recommended by its policies. To assess its performance, an evaluation method is employed, generating corresponding rewards. The process involves key steps in reinforcement learning. The state, denoted as S, signifies the agent's current position within the Kubernetes cluster. Actions (A) represent decisions made by Smart Scheduling, including pod scheduling and scaling. Each action yields a reward, determined by sustainability, performance, and other factors, weighted accordingly (`c_sus`, `c_perf`, `c_res`, `w_sus`, `w_perf`, `w_res`). Episodes conclude when the agent reaches a terminating state, unable to take further actions.
+
+Temporal Difference, a specific formula, calculates the Q-Value, gauging the effectiveness of an action (A) taken at a given state (S). This value, denoted as `Q(A, S)`, is iteratively updated using the Bellman equation and temporal differences. The objective is to minimize state transfer costs, optimizing Smart Scheduling’s decisions within the Kubernetes cluster.
+
+![Typical Scheduling Policy](images/241-test-result.png)
```

Comment:
```suggestion
<p class="mt-5 mb-5"><img src="/images/blogs/2023-09-cloud-native-sustainability-week/241-test-result.png"></p>
```

@jiangphcn could you move the image to `/images/blogs/2023-09-cloud-native-sustainability-week/241-test-result.png`